docs/src/main/tut/docs/cassandra/README.md

```diff
@@ -4,6 +4,265 @@ title: Cassandra
+      index: Int,
+      value: T,
+      codec: ByteBufferCodec[T]): FS[BoundStatement]
+
+  def setValueByName[T](
+      boundStatement: BoundStatement,
+      name: String,
+      value: T,
+      codec: ByteBufferCodec[T]): FS[BoundStatement]
+
+  def setByteBufferListByIndex(
+      preparedStatement: PreparedStatement,
+      values: List[SerializableValueBy[Int]]): FS[BoundStatement]
+
+  def setByteBufferListByName(
+      preparedStatement: PreparedStatement,
+      values: List[SerializableValueBy[String]]): FS[BoundStatement]
+}
+```
+
+### ResultSetAPI
+Provides methods to interact directly with a Cassandra ResultSet, so we can get the final Object 
+representation, a list of them, or describe that that ResultSet could not have been found in the 
+current Cassandra instance.
+
+```
+@free
+trait ResultSetAPI {
+
+  def read[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[A]
+
+  def readOption[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[Option[A]]
+
+  def readList[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[List[A]]
+
+}
+```
+
+## Freatures
+
+### Low level Queries
+The Low-Level API allows us to define queries with placeholders for query constraints, which can be 
+later bound to the final values, running the query in a safe mode, preventing CQL injection. 
+Frees-Cassandra checks the generated query is a valid one at compile time, throwing errors in case 
+any input value doesn't match with the expected data type. 
+
+Let's see how we can use it:
+
+```
+for {
+  preparedStatement <- sessionAPI.prepare("INSERT INTO users (id, name) VALUES (?, ?)")
+  boundStatement    <- bindValues(preparedStatement)
+  _                 <- sessionAPI.executeStatement(boundStatement)
+  user <- sessionAPI.executeWithMap(
+    s"SELECT id, name FROM users WHERE id = ?",
+    Map("id" -> newUser.id)) map { rs =>
+    val row = rs.one()
+    User(row.getUUID(0), row.getString(1))
+  }
+} yield user
+
+```
+
+Note that we are using a bind function, which is defined as follows:
+
+```
+def bindValues(st: PreparedStatement)(
+    implicit c1: ByteBufferCodec[UUID],
+    c2: ByteBufferCodec[String]): FreeS[F, BoundStatement] =
+  List(
+    statementAPI.setValueByName(_: BoundStatement, "id", newUser.id, c1),
+    statementAPI.setValueByName(_: BoundStatement, "name", newUser.name, c2))
+    .foldLeft[FreeS[F, BoundStatement]](statementAPI.bind(st)) { (freeS, func) =>
+      freeS.flatMap(boundSt => func(boundSt))
+    }
+```
+
+
+### String Interpolator
+Frees-Cassandra Query Interpolator allows us to write a raw query and validate it against a 
+previously defined schema. 
+
+To do so, we need to first define an annotated trait specifying a path to the file containing the
+cassandra keyspaces & tables schemas. 
+
+Example
+
+```  @SchemaFileInterpolator("/schema.cql")
+     trait DummySchemaInterpolator
+```
+
+This annotation will be expanded via macros to implement a companion object defining a 
```

Comment:
We don't need to specify details as to how the impl of expanding the object works as that may change and this would get outdated. I think here we should just emphasize that you can validate the schema at compile time and runtime and what the user code to do that looks like.